packages/notifications/src/elements/toaster/ToastSlot.spec.tsx

```diff
@@ -35,7 +35,11 @@ describe('ToastSlot', () => {
   it('pauses toast timers when placement is hovered', async () => {
     const { getByRole, getAllByText } = render(<ToastExample />);
 
-    await Promise.all(new Array(5).fill(null).map(() => user.click(getByRole('button'))));
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const _ of new Array(5).fill(null)) {
+      // eslint-disable-next-line no-await-in-loop
+      await user.click(getByRole('button'));
+    }
```

Comment:
The `for of` loop was favorable here to keep the async calls originally done like so:
```
await Promise.all(new Array(5).fill(null).map(() => user.click(getByRole('button'))));
```
as it was throwing a warning on what appeared to be stacking `act` calls being done internally by the `user.click` API.

Reply:
Can this be refactored as an internal `await` function as shown by https://eslint.org/docs/latest/rules/no-await-in-loop?

Reply:
Having a classic for loop as it's shown above is equivalent to:

```
await Promise.all(new Array(5).fill(null).map(() => user.click(getByRole('button'))));
```

Since this mixes a sync invocation to an async operation because all the async operations are running simultaneously. This doesn't guarantee sequence nor waiting for atomic completion per invocation before the next call is started synchronously - this caused warnings to show up. An alternative to `for of` loop that would wait per call is to:

```
await user.click(getByRole('button'));
await user.click(getByRole('button'));
await user.click(getByRole('button'));
await user.click(getByRole('button'));
await user.click(getByRole('button'));
```